### DISCUSSION TITLE
Move the tests directory outside of the code directory?

### DISCUSSION THREAD

**Discussion by Armavica** [pymc-devs/pymc3#6172] [General]
Hi,
With the reorganization of the tests complete (#5777), a last question remains: where to place the `tests` directory. It currently resides inside the `pymc` code directory:

```
pymc/
├── benchmarks/
├── docs/
├── pymc/
│   ├── __init__.py
│   ├── model.py
│   ├── ...
│   └── tests/
└── scripts/
```

but one could argue that it would make more sense to put it at the same level as `benchmarks/`, `docs/` and `pymc/`:

```
pymc/
├── benchmarks/
├── docs/
├── pymc/
│   ├── __init__.py
│   ├── model.py
│   └── ...
├── scripts/
└── tests/
```

This is what is recommended by the [official Python packaging guide](https://packaging.python.org/en/latest/tutorials/packaging-projects/), by the [Pytest good practices guide](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#tests-outside-application-code), the [Poetry package management tool](https://python-poetry.org/docs/basic-usage/) as well as by [other guides](https://docs.python-guide.org/writing/structure/).
However, I have to note that `numpy`, `scipy`, and `pandas` all have `tests` inside the code directory.

Besides the arguments advanced by the [Pytest good practices guide](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#tests-outside-application-code), I can add that I often `grep` the source code directory to see where or how a function or class is used or defined. With the tests directory where it is currently, a lot of noise is often returned from the tests.

Cost of the transition: probably limited to some `import`s to change in the test files to reflect the move.

What is your opinion on this?

**Comment by ricardoV94** [comment]
Sounds fine by me if nobody has objections. 

Some people do seem to like to run all tests when they download a package but I think those are a very small minority.

**Reply by Armavica** [reply]
> Some people do seem to like to run all tests

Would this be problematic with the `tests` directory outside of the code directory?

With `aesara` which follows this structure for example, launching `pytest` in the main directory does seem to run all the tests.

**Reply by ricardoV94** [reply]
I think the tests are no longer included in package installation in those cases.
